Content.Server/_NF/SizeAttribute/SizeAttributeSystem.cs

```diff
@@ -0,0 +1,95 @@
+using System.Numerics;
+using Content.Server.SizeAttribute;
+using Robust.Server.GameObjects;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Collision.Shapes;
+using Robust.Shared.Physics.Systems;
+using Content.Shared.Item.PseudoItem;
+using Content.Shared.Nyanotrasen.Item.PseudoItem;
+
+namespace Content.Server._NF.SizeAttribute
+{
+    public sealed class SizeAttributeSystem : EntitySystem
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+        [Dependency] private readonly AppearanceSystem _appearance = default!;
+        [Dependency] private readonly FixtureSystem _fixtures = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SizeAttributeComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, SizeAttributeComponent component, ComponentInit args)
+        {
+            if (!TryComp<SizeAttributeWhitelistComponent>(uid, out var whitelist))
+                return;
+
+            if (whitelist.Tall && component.Tall)
+            {
+                Scale(uid, component, whitelist.TallScale, whitelist.TallDensity, whitelist.TallCosmeticOnly);
+                PseudoItem(uid, component, whitelist.TallPseudoItem);
+            }
+            else if (whitelist.Short && component.Short)
+            {
+                Scale(uid, component, whitelist.ShortScale, whitelist.ShortDensity, whitelist.ShortCosmeticOnly);
+                // deltav: added PseudoItemShape
+                PseudoItem(uid, component, whitelist.ShortPseudoItem, whitelist.PseudoItemShape);
+            }
+        }
+
+        // deltav: added a shape argument and refactored the method
+        private void PseudoItem(EntityUid uid, SizeAttributeComponent component, bool active, List<Box2i>? shape = null)
+        {
+            if (active)
+            {
+                var pseudoItem = EnsureComp<PseudoItemComponent>(uid);
+                pseudoItem.Shape = shape;
+            }
+            else
+            {
+                if (!TryComp<PseudoItemComponent>(uid, out var pseudoI))
+                    return;
+                RemComp(uid, pseudoI);
+            }
+        }
+
+        private void Scale(EntityUid uid, SizeAttributeComponent component, float scale, float density, bool cosmeticOnly)
+        {
+            if (scale <= 0f && density <= 0f)
```

Comment:
It would probably be better to do 
if (scale <= 0f | density <= 0f) return;
since && only returns if both scale and density aren't a natural number at the same time. Scale could be a negative number and density any natural number, and the function would continue as normal. Which would produce huge problems since we don't want to accidentally mirror people's character models in the case of a negative scale being put in somewhere. Keep in mind that the compiler doesn't actually know that having negative inputs for scale isn't acceptable, since it's technically possible and won't throw any errors. 

Alternatively you could put in a check for if scale is a negative number, and then set it to *= -1 to get a natural number again.